src/main/java/teammates/ui/controller/AdminEmailTrashAction.java

```diff
@@ -17,7 +17,7 @@ protected ActionResult execute() {
         String emailId = getRequestParamValue(Const.ParamsNames.ADMIN_EMAIL_ID);
         
         String redirect = getRequestParamValue(Const.ParamsNames.ADMIN_EMAIL_TRASH_ACTION_REDIRECT);
-        redirect = redirect == null ? Const.ActionURIs.ADMIN_EMAIL_TRASH_PAGE : redirect;
+        redirect = redirect == null ? Const.ActionURIs.ADMIN_EMAIL_TRASH_PAGE : redirect; // NOPMD
```

Comment:
On a side note, 5.3.7 (the version the author said the problem's fixed) is already in the wild, but I'm not sure if it's compatible with 5.4.1.